### PR TITLE
Update docs, migrate cards from `cardImplementations/` to `cards/`

### PR DESCRIPTION
**Pull request by AMMayberry1**
Reworked the contributor's guide to reflect our changes for SWU. Moved all card implementations from `cardImplementations/` folders to `cards/` folders.

Also did some various cleanup of interfaces and card implementations since a lot of this stuff will be the examples that people refer to.

### CODE REVIEW

File: docs/implementing-cards.md [hunk 1]
```diff
@@ -1,388 +1,570 @@
...
 
-### Keywords
+There are several ability types in SWU, each with its own initialization method. As shown above in the example of an action ability, each method accepts a property object defining the ability's behavior. Use the `AbilityHelper` import to get access to tools to help with implementations. Additionally, see [Interfaces.ts](../server/game/Interfaces.ts) for a list of available parameters for each ability type.
 
-Most Keywords are automatically parsed from the card text. It isn't necessary to explicitly implement them unless they are provided by a conditional persistent effect.
+The ability types and methods are:
 
-### Static bonuses from attachments
+| Ability Type | Method | Definition |
+| --- | --- | --- |
```
Comment (adbronico) [inline docs/implementing-cards.md:94]: We should add an example column here with a couple varied examples

File: docs/implementing-cards.md [hunk 2]
```diff
@@ -1,388 +1,570 @@
...
-    match: card => card.isDefending(),
-    targetController: 'opponent',
-    effect: AbilityDsl.effects.modifyMilitarySkill(-1)
+// While this card is in the ground arena, all of the opponent's units in the space arena get -1/-1
+this.constantAbility({
+    sourceLocationFilter: Location.GroundArena,
+    targetLocationFilter: Location.SpaceArena,
+    targetCardType: WildcardCardType.Unit,
+    targetController: RelativePlayer.Opponent,
+    ongoingEffect: AbilityHelper.ongoingEffects.modifyStats({ power: -1, hp: -1 })
```
Comment (adbronico) [inline docs/implementing-cards.md:168]: modifyStats should be 1, 1 not -1, -1

File: docs/implementing-cards.md [hunk 3]
```diff
@@ -1,388 +1,570 @@
...
-    ]
+// Entrenched ability
+this.addConstantAbilityTargetingAttached({
+    title: 'Attached unit cannot attack bases',
+    ongoingEffect: AbilityHelper.ongoingEffects.cannotAttackBase(),
 });
 ```
 
-#### Applying effects to cards which aren't in play
+#### **IGNORE THIS SECTION, STILL WIP**: ~~Applying effects to cards which aren't in play~~
```
Comment (adbronico) [inline docs/implementing-cards.md:229]: Can we update all Ignore this, WIP and TBD type notes in here to all be TODO for future finding and referencing?

File: docs/implementing-cards.md [hunk 4]
```diff
@@ -1,388 +1,570 @@
...
 ```javascript
-this.action({
-    title: 'Sacrifice to discard an attachment'
-    target: {
-        cardType: 'attachment',
-        gameAction: AbilityDsl.actions.discardFromPlay()
-    },
-    // ...
+class AbilityContext {
+    constructor(properties) {
```
Comment (adbronico) [inline docs/implementing-cards.md:478]: Add a TODO to make this look more SWU-ish (this could apply in a few places)

**Review by adbronico — CHANGES_REQUESTED** [review]
This looks really good. There's a few small things but overall this is a great start.

File: docs/implementing-cards.md (hunk 1 above)
Comment (AMMayberry1) [reply docs/implementing-cards.md:94]: Do you mean examples of abilities or examples of code implementations?

File: docs/implementing-cards.md (hunk 2 above)
Comment (AMMayberry1) [reply docs/implementing-cards.md:168]: Why is that? It's just a random example, and the comment says -1/-1

File: docs/implementing-cards.md (hunk 4 above)
Comment (AMMayberry1) [reply docs/implementing-cards.md:478]: Can you elaborate? What would make the context class more SWUish?

File: docs/implementing-cards.md (hunk 1 above)
Comment (adbronico) [reply docs/implementing-cards.md:97]: @AMMayberry1 just like example cards for each one so people would know what to look

File: docs/implementing-cards.md (hunk 2 above)
Comment (adbronico) [reply docs/implementing-cards.md:171]: I must have misread the comment as 1/1. My b

File: docs/implementing-cards.md (hunk 4 above)
Comment (adbronico) [reply docs/implementing-cards.md:481]: @AMMayberry1 Stuff like rings and stage being removed for example. It isn't high priority, but someone might assume this is an accurate breakdown of the context object when they're new.

Comment (AMMayberry1) [reply docs/implementing-cards.md:481]: Oh lol right I don't even notice ring stuff anymore, good catch I'll remove it.

I think Stage might actually still be relevant (I think it means "ability resolution stage") but we're not using it in many places. I'll add a TODO in the code for later investigation to see if we still need it.

**Review by adbronico — APPROVED** [review]
Merge or merge not, there is no try

**Pull request merged**